phpBB/phpbb/avatar/driver/upload.php

```diff
@@ -73,7 +86,7 @@ public function get_data($row)
 		$root_path = (defined('PHPBB_USE_BOARD_URL_PATH') && PHPBB_USE_BOARD_URL_PATH) ? generate_board_url() . '/' : $this->path_helper->get_web_root_path();
 
 		return array(
-			'src' => $root_path . 'download/file.' . $this->php_ext . '?avatar=' . $row['avatar'],
+			'src' => $root_path . $this->controller_helper->route('phpbb_storage_avatar', ['file' => $row['avatar']]),
```

Comment:
Just tested this and noticed that due to using `controller_helper->route()` we no longer need the `$root_path`
```suggestion
			'src' => $this->controller_helper->route('phpbb_storage_avatar', ['file' => $row['avatar']]),
```
... and remove the `$root_path` above.

Reply:
Otherwise the avatars are not properly displayed and actually broken ;)